R/proDA.R

```diff
@@ -313,36 +313,46 @@ fit_parameters_loop <- function(Y, model_matrix, location_prior_df,
   }
   # Initialization
   n_samples <- ncol(Y)
-
-
+  mM_t <- t(model_matrix)
+  seq_Y <- seq_len(nrow(Y))
+  seq_mM <- seq_len(nrow(model_matrix))
+  
   Y_compl <- Y
-  Y_compl[is.na(Y)] <- rnorm(sum(is.na(Y)), mean=quantile(Y, probs=0.1, na.rm=TRUE), sd=sd(Y,na.rm=TRUE)/5)
-  res_init <- lapply(seq_len(nrow(Y)), function(i){
-    pd_lm.fit(Y_compl[i, ], model_matrix,
+  y_na <- is.na(Y)
+  Y_compl[y_na] <- rnorm(sum(y_na), mean=quantile(Y, probs=0.1, na.rm=TRUE), sd=sd(Y,na.rm=TRUE)/5)
+  
+  split_Y_compl <- split(Y_compl, factor(rownames(Y_compl), rownames(Y_compl)))
```

Comment:
Why do you pass `rownames(Y_compl)` to factor twice?

What happens if the rownames are not unique? (I know one shouldn't make objects with duplicated rownames, but you can, which means that people probably do.)

Reply:
``` r
mat <- matrix(1:20, nrow=5, ncol = 4)
rownames(mat) <- paste0("row_", 1:5)
rownames(mat)[3] <- "row_2"
split(mat, factor(rownames(mat)))
#> $row_1
#> [1]  1  6 11 16
#> 
#> $row_2
#> [1]  2  3  7  8 12 13 17 18
#> 
#> $row_4
#> [1]  4  9 14 19
#> 
#> $row_5
#> [1]  5 10 15 20
```

<sup>Created on 2021-01-27 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
Can you please validate that split is actually faster than `lapply()`, because it makes the code more difficult to understand and maintain.

Reply:
Agreed, maybe it's not a good idea to use `split` here given the points you addressed. I will revert this.